### PR TITLE
Rename h3_edge to get_edge

### PR DESCRIPTION
**Pull request by madninja**
Fix up result check of getH3UnidirectionalEdge to check for 0 instead
of making another call

### CODE REVIEW

**Comment by madninja** [comment]
@christianjgreen this renames `h3_edge` to `get_edge` to match naming convention. I also made checking the result of `getH3UnidirectionalEdge ` simpler since the function already returns either a valid index or 0 if either of the two indices was invalid or not at the same resolution. 

I _think_ that throwing a `badarg` is still valid here since, if the arguments are not at the same resolution, the caller passed in invalid arguments.  I think we should keep your comment in mind for error responses though and evaluate carefully what the right setup should be.   

For example this PR also fixes up the just merged grid_distance function to throw a badarg since the caller passed in to indices that are not compatible or separated by penthagonal distortion. While we could construct an error tuple, I'm not sure it would help much. 

Open to revisiting error responses at some point though

**Review by vihu — COMMENTED** [review]
Honestly I’d name it get_unidirectional_edge. Unless there’s no other edge function in h3, in that case it’s fine as get_edge.

**Comment by christianjgreen** [comment]
First, thanks for the quick review and feedback!

I think it would be best to keep `unidirectional` in the function name to distinguish from potential "edge" functions down the pipe. There is an issue open talking about a potential "bidirectional edge": https://github.com/uber/h3/issues/133

File: c_src/h3.c [hunk 1]
```diff
@@ -633,11 +633,16 @@ erl_grid_distance(ErlNifEnv * env, int argc, const ERL_NIF_TERM argv[])
     }
 
     int64_t result = h3Distance(h3idx_src, h3idx_dest);
+    if (result < 0)
```
Comment (christianjgreen) [inline c_src/h3.c:636]: Nice catch! If I get around to the next batch of edge functions, I think it could be useful to have an h3 _edge_ helper function.

**Comment by madninja** [comment]
> First, thanks for the quick review and feedback!
> 
> I think it would be best to keep `unidirectional` in the function name to distinguish from potential "edge" functions down the pipe. There is an issue open talking about a potential "bidirectional edge": [uber/h3#133](https://github.com/uber/h3/issues/133)

Done

File: c_src/h3.c (hunk 1 above)
Comment (madninja) [reply c_src/h3.c:636]: Yeah, when we get to it, helpers will keep the code DRY

**Comment by christianjgreen** [comment]
Wow you guys don't mess around, that was quick! Can't approve, but looks good to me!

**Comment by madninja** [comment]
Heh, no messin' around.. We happen to need that grid_distance function in another project, so I figured I may as well pick your stuff up before hex'ing another version

**Comment by madninja** [comment]
@christianjgreen also pushed to hex.pm as v3.1.0 fyi

**Pull request merged**